tests/model/eventsources/test_api_event_source.py

```diff
@@ -21,10 +22,16 @@ def setUp(self):
 
         self.stage = "Prod"
         self.suffix = "123"
+        self.kwargs = {
+            "function": self.func,
+            "explicit_api": {},
+            "api_id": "RestApi",
+            "intrinsics_resolver": IntrinsicsResolver({}),
```

Comment:
what's the reason for messing with kwargs here (can't deduce from the code) rather than testing it how it was before?

Reply:
This is because we changed the way we get the `intrinsics_resolver` in the `to_cloudformation()` method from `intrinsics_resolver = kwargs.get("intrinsics_resolver")` to `intrinsics_resolver: IntrinsicsResolver = kwargs["intrinsics_resolver"]` and caused those tests to fail because we now expect there to always be an `IntrinsicsResolver`. We changed this because the `self.add_auth_to_swagger()` call compalins about `intrinsics_resolver` being returned as `Optional` by the `.get()` method. It is returned similarly in the `sam_resources.py` file.

Reply:
So we're doing that only because of the static type check?
Then we should initialize `intrinsic_resolver` to an empty object in `push.py` and don't touch the tests as it can be a backward compat issue

Reply:
> We changed this because the self.add_auth_to_swagger() call compalins about intrinsics_resolver being returned as Optional by the .get() method.

what was complaining exactly? (if you have the err msg that'd be helpful)